### PR TITLE
v2alpha1: Add vesting, vault and drain vault contents to Transaction

### PR DESCRIPTION
**Pull request by kacpersaw**

### CODE REVIEW

File: spacemesh/v2alpha1/tx.proto [hunk 1]
```diff
@@ -22,10 +22,13 @@ message Transaction {
     TRANSACTION_TYPE_UNSPECIFIED = 0;
     TRANSACTION_TYPE_SINGLE_SIG_SEND = 1;
     TRANSACTION_TYPE_SINGLE_SIG_SPAWN = 2;
-    TRANSACTION_TYPE_SINGLE_SIG_SELFSPAWN = 3;
+    TRANSACTION_TYPE_SINGLE_SIG_SELFSPAWN = 3; // deprecated
```
Comment (poszu) [inline spacemesh/v2alpha1/tx.proto:25]: How about using [deprecated option](https://protobuf.dev/programming-guides/proto3/#options)?
```suggestion
    TRANSACTION_TYPE_SINGLE_SIG_SELFSPAWN = 3 [deprecated = true];
```

File: spacemesh/v2alpha1/tx.proto [hunk 2]
```diff
@@ -45,11 +48,28 @@ message ContentsMultiSigSpawn {
   repeated string pubkey = 2;
 }
 
+message ContentsVaultSpawn {
+  string owner = 1;
+  uint64 total_amount = 2;
+  uint64 initial_unlock_amount = 3;
+  uint32 vesting_start = 4;
+  uint32 vesting_end = 5;
```
Comment (poszu) [inline spacemesh/v2alpha1/tx.proto:56]: What kind of units are these?

File: spacemesh/v2alpha1/tx.proto (hunk 1 above)
Comment (kacpersaw) [reply spacemesh/v2alpha1/tx.proto:25]: Cool, I didn't know that

File: spacemesh/v2alpha1/tx.proto (hunk 2 above)
Comment (kacpersaw) [reply spacemesh/v2alpha1/tx.proto]: Layer number - added comments

**Review by brusherru — APPROVED** [review]

**Review by poszu — APPROVED** [review]

**Pull request merged**